File: core/src/main/java/com/views/calendar/cosmocalendar/settings/selection/SelectionModel.java
package com.views.calendar.cosmocalendar.settings.selection;

import com.views.calendar.cosmocalendar.utils.SelectionType;

public class SelectionModel implements SelectionInterface {

    //Selecton type SINGLE, MULTIPLE, RANGE, NONE
    @SelectionType
    private int selectionType;

    @Override
    @SelectionType
    public int getSelectionType() {
        return selectionType;
    }

    @Override
    public void setSelectionType(@SelectionType int selectionType) {
        this.selectionType = selectionType;
    }
}

File: core/src/main/java/com/views/calendar/cosmocalendar/settings/selection/SelectionModel.kt
package com.views.calendar.cosmocalendar.settings.selection

import com.views.calendar.cosmocalendar.utils.SelectionType

class SelectionModel : SelectionInterface {
    // Selecton type SINGLE, MULTIPLE, RANGE, NONE
    @get:SelectionType
    @SelectionType
    override var selectionType = 0
}
